### PR TITLE
[mlir] Add `res()` method to `linalg::ContractionOpInterface`

### PR DESCRIPTION
**Pull request by pzread**
In addition to `lhs()` and `rhs()` to return left and right operands, add `res()` to return the result value.

### CODE REVIEW

File: mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td [hunk 1]
```diff
@@ -54,6 +54,14 @@ def LinalgContractionOpInterface : OpInterface<"ContractionOpInterface"> {
       return $_op.getOperation()->getOperand(1);
     }]>,
     InterfaceMethod<
+    /*desc=*/"Returns the result value.",
+    /*retTy=*/"OpResult",
+    /*methodName=*/"res",
+    /*args=*/(ins),
+    /*methodBody=*/[{
+      return $_op.getOperation()->getResult(1);
```
Comment (joker-eph) [inline mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td:62]: What is result 0?

**Review by joker-eph — COMMENTED** [review]
This will be dead code, can you add tests for this?

File: mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td (hunk 1 above)
Comment (pzread) [reply mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td]: Ah, it should be 0. A unit test would catch this : ) Fixed.

**Comment by pzread** [comment]
> This will be dead code, can you add tests for this?

Added the unit test

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**